lib/Image/Image.js

```diff
@@ -114,46 +123,51 @@ module.exports = kind(
 	/**
 	* @private
 	*/
-	defaultKind: ImageBadge,
-
-	/**
-	* @private
-	*/
 	components: [
-		{name: 'image', kind: Img},
-		{name: 'client', kind: Control, canGenerate: false, classes: 'moon-image-client'}
+		{name: 'image', kind: Img}
 	],
 
 	/**
 	* @private
 	*/
 	bindings: [
-		{from: '.src', to: '.$.image.src'},
-		{from: '.alt', to: '.$.image.alt'},
-		{from: '.sizing', to: '.$.image.sizing'},
-		{from: '.position', to: '.$.image.position'}
+		{from: 'src', to: '$.image.src'},
+		{from: 'alt', to: '$.image.alt'},
+		{from: 'sizing', to: '$.image.sizing'},
+		{from: 'position', to: '$.image.position'}
 	],
 
 	/**
-	* Only generate `this.$.client` if the instance has components.
-	*
-	* @method
 	* @private
 	*/
 	create: function () {
+		this.adaptComponentsBlock();
 		Control.prototype.create.apply(this, arguments);
-		if (this.getClientControls().length > 0) {
-			this.$.client.canGenerate = true;
-			this.addClass('has-children');
-		}
+	},
 
-		this.showBadgesOnSpotlightChanged();
+	/**
+	* Adapting from the prior API. We'll assume that the presence of components but not
+	* overlayComponents means the consumer is expecting the former ImageBadge-style overlay.
+	*
+	* @deprecated Backwards compatibility for moon.ImageBadge's within components block
+	* @private
+	*/
+	adaptComponentsBlock: function () {
+		if (this.components && !this.overlayComponents) {
+			this.overlayComponents = this.components;
+			this.components = null;
+			this.overlayComponents.forEach(function (c) {
```

Comment:
From a perf standpoint, would it be faster to use a standard `for` loop, semantics aside?